src/core/radio/trel_interface.cpp

```diff
@@ -34,101 +34,415 @@
+    }
+
+    if (entry == nullptr)
+    {
+        entry = mPeerTable.FindMatching(extAddress);
+    }
+
+    if (entry == nullptr)
+    {
+        entry = GetNewPeerEntry();
+        VerifyOrExit(entry != nullptr);
+
+        entry->SetExtAddress(extAddress);
+        isNew = true;
+    }
+
+    if (!isNew)
+    {
+        VerifyOrExit((entry->GetExtPanId() != extPanId) || (entry->GetSockAddr() != aInfo.GetSockAddr()));
+    }
+
+    entry->SetExtPanId(extPanId);
+    entry->SetSockAddr(aInfo.GetSockAddr());
+
+    entry->Log(isNew ? "Added" : "Updated");
 
 exit:
     return;
 }
 
-Error Interface::Send(const Packet &aPacket)
+template <uint16_t kBufferSize>
+static Error ParseValueAsHexString(const Dns::TxtEntry &aTxtEntry, uint8_t (&aBuffer)[kBufferSize])
+{
+    // Parse the value from `Dns::TxtEntry` as a hex string and
+    // populates the parsed bytes into `aBuffer`. It requires parsing of
+    // the hex string to result in exactly `kBufferSize` decoded bytes.
+
+    Error error = kErrorParse;
+    char  hexString[kBufferSize * 2 + 1];
+
+    VerifyOrExit(aTxtEntry.mValueLength < sizeof(hexString));
+
+    memcpy(hexString, aTxtEntry.mValue, aTxtEntry.mValueLength);
+    hexString[aTxtEntry.mValueLength] = '\0';
+
+    error = Utils::CmdLineParser::ParseAsHexString(hexString, aBuffer);
+
+exit:
+    return error;
+}
+
+Error Interface::PrasePeerInfoTxtData(const Peer::Info &  aInfo,
+                                      Mac::ExtAddress &   aExtAddress,
+                                      Mac::ExtendedPanId &aExtPanId) const
+{
+    Error                   error;
+    Dns::TxtEntry           entry;
+    Dns::TxtEntry::Iterator iterator;
+    bool                    parsedExtAddress = false;
+    bool                    parsedExtPanId   = false;
+
+    aExtPanId.Clear();
+
+    iterator.Init(aInfo.GetTxtData(), aInfo.GetTxtLength());
+
+    while ((error = iterator.GetNextEntry(entry)) == kErrorNone)
+    {
+        if (strcmp(entry.mKey, kTxtRecordExtAddressKey) == 0)
+        {
+            VerifyOrExit(!parsedExtAddress, error = kErrorParse);
+            SuccessOrExit(error = ParseValueAsHexString(entry, aExtAddress.m8));
+            parsedExtAddress = true;
+        }
+        else if (strcmp(entry.mKey, kTxtRecordExtPanIdKey) == 0)
+        {
+            VerifyOrExit(!parsedExtPanId, error = kErrorParse);
+            SuccessOrExit(error = ParseValueAsHexString(entry, aExtPanId.m8));
+            parsedExtPanId = true;
+        }
+
+        // Skip over and ignore any unknown keys.
+    }
+
+    VerifyOrExit(error == kErrorNotFound);
+    error = kErrorNone;
+
+    VerifyOrExit(parsedExtAddress, error = kErrorParse);
+
+exit:
+    return error;
+}
+
+Interface::Peer *Interface::GetNewPeerEntry(void)
+{
+    Peer *peerEntry;
+
+    peerEntry = mPeerTable.PushBack();
+    VerifyOrExit(peerEntry == nullptr);
+
+    for (Peer &entry : mPeerTable)
+    {
+        if (entry.GetExtPanId() != Get<Mac::Mac>().GetExtendedPanId())
```

Comment:
We want to allow the new peer from another PAN since it may be a new device that is trying to join (going through Thread `Join` process and it may not yet know the xpan). With current model, we practically allow one entry for such new devices (when table is full).

Reply:
SGTM to prefer new device to older device.